src/snovault/elasticsearch/simple_queue.py

```diff
@@ -0,0 +1,156 @@
+            # There are uuids on the server and the worker is not working
+            while cnt > 0:
+                uuid = self._get_uuid()
+                if not uuid:
+                    break
+                uuids.append(uuid)
+                cnt -= 1
+            client_conn['uuid_cnt'] = len(uuids)
+        return uuids
+
+    def has_uuids(self):
+        '''Are there uuids in the queue'''
+        if self._uuids:
+            return True
+        return False
+
+    def is_indexing(self):
+        '''Is an indexing process currently running'''
+        if self.has_uuids() or self.errors:
+            return True
+        for client_info in self._client_conns.values():
+            if client_info['uuid_cnt']:
+                return True
+            elif client_info['results'] is None:
+                return True
+        return False
+
+    def load_uuids(self, uuids):
+        '''
+        The only way to load uuids in queue
+        - Uuids are to be loaded once per indexing session
+        '''
+        if self.has_uuids():
+            return None
+        for uuid in uuids:
+            self._load_uuid(uuid)
+        return len(self._uuids)
+
+    def reset(self, force=False):
+        '''
+        Reset the queue so a new indexing process can begin
+        - There cannot be any uuids remaining
+        - The errors from the last indexing session must be removed
+        - There cannot be any pending client connections
+        '''
+        msg = 'Okay'
+        can_reset = True
+        if self.has_uuids():
+            can_reset = False
+            msg = 'Queue Server cannot reset. Uuids=%d' % (
+                len(self._uuids),
+            )
+        if self.errors:
+            can_reset = False
+            msg = 'Queue Server cannot reset. Errors=%d' % (
+                len(self.errors),
+            )
+        for client_id, client_info in self._client_conns.items():
+            if client_info['uuid_cnt']:
+                can_reset = False
+                msg = 'Queue Server cannot reset. Client Pending(%s)' % (
+                    client_id,
+                )
+        if can_reset or force:
+            self._uuids = []
+            self._client_conns = {}
+            self.errors = []
+        return msg
+
+    def update_finished(self, client_id, results):
+        '''The way a client finishes a batch of uuids'''
+        msg = 'Client id(%s) DNE.' % client_id
+        if client_id in self._client_conns:
+            client_conn = self._client_conns[client_id]
+            client_conn['results'] = results
+            batch_errors = []
+            for error in results['errors']:
+                error['client_id'] = client_id
+                batch_errors.append(error)
+            self.errors.extend(batch_errors)
+            client_conn['uuid_cnt'] -= results['successes']
+            client_conn['uuid_cnt'] -= len(batch_errors)
+            if results['successes'] < 0 or client_conn['uuid_cnt']:
```

Comment:
?:

msg = 'Queue Server cannot close client(%s)' % client_id
return  msg if results['successes'] < 0 or client_conn['uuid_cnt'] else 'Okay'

Reply:
Haha, you like this these in-line conditionals...When I'm in a hurry I'll just set the var in long form conditionals and return at the end.  Yea, could be cleaned up for sure.